src/TheThingsNetwork.cpp

```diff
@@ -337,3 +250,148 @@ void TheThingsNetwork::showStatus() {
   debugPrint(F("RX Delay 2: "));
   debugPrintLn(readValue(F("mac get rxdelay2")));
 }
+
+void TheThingsNetwork::configure_EU868(int sf) {
+  int ch;
+  int dr = -1;
+  long int freq = 867100000;
+  String str = "";
+
+  str.concat(F("mac set rx2 3 869525000"));
+  sendCommand(str);
+  str = "";
+  for (ch = 0; ch <= 7; ch++) {
+    if (ch >= 3) {
+      str.concat(F("mac set ch freq "));
+      str.concat(ch);
+      str.concat(F(" "));
+      str.concat(freq);
+      sendCommand(str);
+      str = "";
+      str.concat(F("mac set ch drrange "));
+      str.concat(ch);
+      str.concat(F(" 0 5"));
+      sendCommand(str);
+      str = "";
+      str.concat(F("mac set ch status "));
+      str.concat(ch);
+      str.concat(F(" on"));
+      sendCommand(str);
+      str = "";
+      freq = freq + 200000;
+    }
+    str.concat(F("mac set ch dcycle "));
+    str.concat(ch);
+    str.concat(F(" 799"));
+    sendCommand(str);
+    str = "";
+  }
+  str.concat(F("mac set ch drrange 1 0 6"));
+  sendCommand(str);
+  str = "";
+  str.concat(F("mac set pwridx "));
+  str.concat(TTN_PWRIDX_868);
+  sendCommand(str);
+  switch (sf) {
+    case 7:
+      dr = 5;
+      break;
+    case 8:
+      dr = 4;
+      break;
+    case 9:
+      dr = 3;
+      break;
+    case 10:
+      dr = 2;
+      break;
+    case 11:
+      dr = 1;
+      break;
+    case 12:
+      dr = 0;
+      break;
+    default:
+      debugPrintLn(F("Invalid SF"))
+      break;
+  }
+  if (dr > -1){
+    str = "";
+    str.concat(F("mac set dr "));
+    str.concat(dr);
+    sendCommand(str);
+  }
+}
+
+void TheThingsNetwork::configure_US915(int sf, int fsb) {
+  int ch;
+  int dr = -1;
+  String str = "";
+  int chLow = fsb > 0 ? (fsb - 1) * 8 : 0;
+  int chHigh = fsb > 0 ? chLow + 7 : 71;
+  int ch500 = fsb + 63;
+
+  sendCommand(F("radio set freq 904200000"));
+  str = "";
+  str.concat(F("mac set pwridx "));
+  str.concat(TTN_PWRIDX_915);
+  sendCommand(str);
+  for (ch = 0; ch < 72; ch++) {
+    str = "";
+    str.concat(F("mac set ch status "));
+    str.concat(ch);
+    (ch == ch500 || ch <= chHigh && ch >= chLow) ? str.concat(F(" on")) : str.concat(F(" off"));
+    sendCommand(str);
+    if (ch < 63) {
```

Comment:
I was wondering if it was better if we would only initialize the channels that are 'on' and do this:
``` if (ch < 63 && ch == (ch500 || ch <= chHigh && ch >= chLow)) ```